feat(OH-188): 계정 관리 API 개발

rhrudska987 opened the pull request:
## 관련 이슈
https://onehunnit.atlassian.net/browse/OH-188

## 작업 내용
- 계정 수정
닉네임, 생일, 휴대전화, 성별, 계정 상태, 프로필 이미지 수정 가능

- 계정 탈퇴 
회원: 모든 기록 삭제(일기, 위치, 안전구역 등)
보호자: 환자와의 관계 삭제

## 참고사항

rhrudska987 committed:
feat(OH-188): 계정 관리 API 개발

## 관련 이슈
https://ureca.atlassian.net/browse/UR-188

## 작업 내용
- 계정 수정
닉네임, 생일, 휴대전화, 성별, 계정 상태, 프로필 이미지 수정 가능
- 계정 탈퇴
회원: 모든 기록 삭제(일기, 위치, 안전구역 등)
보호자: 환자와의 관계 삭제

sjsjsj1246 commented on src/main/java/kr/co/onehunnit/onhunnit/config/WebSocketHandler.java:
```diff
@@ -42,6 +58,13 @@ protected void handleTextMessage(WebSocketSession session, TextMessage message)
 		locationRepository.save(location);
 	}
 
+	boolean isNotContainAllField(JSONObject jsonObject) {
+		if (jsonObject.containsKey("patientId") && jsonObject.containsKey("longitude") && jsonObject.containsKey("latitude")) {
+			return false;
+		}
+		return true;
+	}
```
이런 함수들은 주의할게. 부정적인? 함수면 헷갈릴 가능성이 커요. 그래서 isContainAllField가 좋을 것 같아요.
그리고 좀더 클린하게
```suggestion
boolean isContainAllField(JSONObject jsonObject) {
    return jsonObject.containsKey("patientId") && jsonObject.containsKey("longitude") && jsonObject.containsKey("latitude");
}
```

sjsjsj1246 commented on src/main/java/kr/co/onehunnit/onhunnit/config/WebSocketHandler.java:
```diff
@@ -22,14 +22,30 @@ public class WebSocketHandler extends TextWebSocketHandler {
+		try {
+			obj = jsonParser.parse(message.getPayload());
+		} catch (Exception e) {
+			throw new IllegalArgumentException("메시지 형식이 올바르지 않습니다.");
```
또 궁금한거는 에러를 throw하면 어떻게 핸들링되고 있나요?
또 사용자에게는 아무 메세지도 보내주지 않나요?

sjsjsj1246 commented on src/main/java/kr/co/onehunnit/onhunnit/repository/AccountRepository.java:
```diff
@@ -14,8 +14,6 @@ public interface AccountRepository extends JpaRepository<Account, Long> {
 
 	Optional<Account> findByEmailAndProvider(String email, Provider provider);
 
-	boolean existsAccountByEmailAndProvider(String email, Provider provider);
```
👍
혹시나 제가 놓친 사용하고 있지 않은 코드들 함수들이 있으면 다 제거해주세용

rhrudska987 replied:
넵!

sjsjsj1246 commented:
지금까지 Assignees를 안넣어주셨네요? 앞으로는 달아주세요~ 경남 님이 작업 하신거니까요

rhrudska987 committed:
refactor(OH-188): 계정 관리 API 개발

## 관련 이슈
https://onehunnit.atlassian.net/browse/OH-188

## 작업 내용
- 에러 발생 시 에러 종류에 따라 프론트에게 메시지 전달
- WebSocketHandler 함수 분리

sjsjsj1246 reviewed (approved):
LGTM 👍
